src/Eventso.Subscription.Hosting/PoisonEventRetryingHost.cs

```diff
@@ -0,0 +1,111 @@
+                .Select(x => CreateTopicRetryingService(x, pipelineFactory, handlersRegistry, poisonEventStore, deadLetterQueueScopeFactory, loggerFactory))
+                .ToArray();
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            if (_topicRetryingServices.Count == 0)
+            {
+                await Task.Delay(Timeout.InfiniteTimeSpan, stoppingToken);
+                return;
+            }
+            
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                // TODO hide postgres behind some interface
+                await using var connection = _connectionFactory.ReadWrite();
+                await DistributedMonitor.TryEnter(connection, RetryLockId, stoppingToken);
+
+                try
+                {
+                    foreach (var topicRetryingService in _topicRetryingServices)
+                        await topicRetryingService.Retry(stoppingToken);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, $"Dead letter queue retrying failed.");
+                }
+                finally
+                {
+                    await DistributedMonitor.Exit(connection, RetryLockId, stoppingToken);
+                }
+
+                await Task.Delay(TimeSpan.FromMinutes(1), stoppingToken); // TODO get from configuration
+            }
+        }
+
+        private static TopicRetryingService CreateTopicRetryingService(
+            SubscriptionConfiguration config,
+            IMessagePipelineFactory messagePipelineFactory,
+            IMessageHandlersRegistry handlersRegistry,
+            IPoisonEventStore poisonEventStore,
+            IDeadLetterQueueScopeFactory deadLetterQueueScopeFactory,
+            ILoggerFactory loggerFactory)
+        {
+            IEventHandler<Event> eventHandler = new RetryingEventHandler(
+                new Observing.EventHandler<Event>(
+                    handlersRegistry,
+                    messagePipelineFactory.Create(config.HandlerConfig)),
+                deadLetterQueueScopeFactory,
+                poisonEventStore);
+
+            if (config.BatchProcessingRequired)
+                eventHandler = config.BatchConfiguration.HandlingStrategy switch
+                {
+                    BatchHandlingStrategy.SingleType => eventHandler,
+                    BatchHandlingStrategy.SingleTypeLastByKey => new SingleTypeLastByKeyEventHandler<Event>(eventHandler),
+                    BatchHandlingStrategy.OrderedWithinKey => new OrderedWithinKeyEventHandler<Event>(eventHandler),
+                    BatchHandlingStrategy.OrderedWithinType => new OrderedWithinTypeEventHandler<Event>(eventHandler),
+                    _ => throw new InvalidOperationException(
+                        $"Unknown handling strategy: {config.BatchConfiguration.HandlingStrategy}")
+                };
```

Comment:
do you like this copy-paste switch?) I think you should extract handler creation logic somewhere